index.bs

```diff
@@ -6145,6 +6282,358 @@ However, [=authenticators=] that do not utilize [[!FIDO-CTAP]] do not necessaril
+
+#### Registration (`create()`) #### {#sctn-device-publickey-extension-verification-create}
+
+If the `devicePubKey` extension was included on a {{CredentialsContainer/create()|navigator.credentials.create()}} call, then the below verification steps are performed in the context of <a href=#reg-ceremony-verify-extension-outputs>this step</a> of [[#sctn-registering-a-new-credential]] using these variables established therein: |credential|, |clientExtensionResults|, |authData|, and |hash|.
+
+1. Let |attObjForDevicePublicKey| be the value of the `devicePubKey` member of |clientExtensionResults|.
+
+1. Verify that |attObjForDevicePublicKey| is valid CBOR conforming to the syntax defined above and perform CBOR decoding on it to extract the contained fields: |aaguid|, |dpk|, |scope|, |nonce|, |fmt|, |attStmt|.
+    Note: The latter |attObjForDevicePublicKey| fields are referenced exclusively in the below steps and are not to be confused with other fields with the same names in other portions of [=authenticator data=].
+
+1. Verify that {{AuthenticationExtensionsDevicePublicKeyOutputs/signature}} is a valid signature over the [=assertion signature=] [input](#fig-signature) by the [=device public key=] |dpk|. (The signature algorithm is the same as for the [=user credential=].)
+
+1. Optionally, if attestation was requested and the RP wishes to verify it, verify that |attStmt| is a correct [=attestation statement=], conveying a valid [=attestation signature=], by using the [=attestation statement format=] |fmt|'s [=verification procedure=] given |attStmt|. See [[#sctn-device-publickey-attestation-calculations]].
+
+    Note: If |fmt|'s value is "[=none=]" there is no attestation signature to verify.
+
+1. Store the |aaguid|, |dpk|, |scope|, |fmt|, |attStmt| values indexed to the <code>|credential|.{{Credential/id}}</code> in the [=user account=].
+
+Issue(w3c/webauthn#1711): Storing this information ought to only occur if the overall registration operation is successful, i.e., crucially, the "encompassing signature" verified correctly.
+
+See also [[#sctn-device-publickey-extension-usage]] for further details.
+
+#### Authentication (`get()`) #### {#sctn-device-publickey-extension-verification-get}
+
+If the `devicePubKey` extension was included on a {{CredentialsContainer/get()|navigator.credentials.get()}} call, then the below verification steps are performed in the context of <a href=#authn-ceremony-verify-extension-outputs>this step</a> of [[#sctn-verifying-assertion]] using these variables established therein: |credential|, |clientExtensionResults|, |authData|, and |hash|.
```

Comment:
neither "authData" nor "hash" are referenced in our description of the verification.
Do they have to be verified? If not: should we remove them from the list above?

Reply:
They have to be verified, but they are instead referred to as "the assertion signature input" in step 3. They are described like that in the signing steps in "Authenticator extension processing" too. I'm not sure if it's better to keep that reference to not duplicate definitions, or spell out "authData || hash" to be explicit. The signature input seems unlikely to change, so perhaps the latter?

Reply:
bb6424a65cc015c6905febd720825d3133115d61

Reply:
Thanks @agl, perhaps that change should be applied to the authentication verification procedure as well?